backend/api/EventHandlers/MissionScheduler.cs

```diff
@@ -47,45 +47,50 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                         continue;
                     }
 
-                    bool startedSuccessfull = await StartMission(queuedMission);
-                    if (!startedSuccessfull)
+                    try
                     {
-                        var newStatus = MissionStatus.Failed;
+                        await StartMission(queuedMission);
+                    }
+                    catch (MissionException e)
+                    {
+                        const MissionStatus NewStatus = MissionStatus.Failed;
                         _logger.LogWarning(
-                            "Mission {id} was not started successfully. Status updated to '{status}'",
+                            "Mission {id} was not started successfully. Status updated to '{status}'.\nReason: {failReason}",
                             queuedMission.Id,
-                            newStatus
+                            NewStatus,
+                            e.Message
                         );
-                        queuedMission.Status = newStatus;
+                        queuedMission.Status = NewStatus;
+                        queuedMission.StatusReason = $"Failed to start: '{e.Message}'";
                         await MissionService.Update(queuedMission);
                     }
                 }
                 await Task.Delay(_timeDelay, stoppingToken);
             }
         }
 
-        private async Task<bool> StartMission(Mission scheduledMission)
+        private async Task StartMission(Mission queuedMission)
         {
             try
             {
                 var result = await RobotController.StartMission(
-                    scheduledMission.Robot.Id,
-                    scheduledMission.Id
+                    queuedMission.Robot.Id,
+                    queuedMission.Id
                 );
                 if (result.Result is not OkObjectResult)
                 {
-                    throw new MissionException(
-                        result?.Result?.ToString() ?? "Unknown error from robot controller"
-                    );
+                    string errorMessage = "Unknown error from robot controller";
+                    if (result.Result is ObjectResult returnObject)
+                        errorMessage = returnObject.Value?.ToString() ?? errorMessage;
+                    throw new MissionException(errorMessage);
                 }
-                _logger.LogInformation("Started mission '{id}'", scheduledMission.Id);
+                _logger.LogInformation("Started mission '{id}'", queuedMission.Id);
             }
             catch (MissionException e)
             {
-                _logger.LogError(e, "Failed to start mission '{id}'", scheduledMission.Id);
-                return false;
+                _logger.LogError(e, "Failed to start mission '{id}'", queuedMission.Id);
```

Comment:
We log this excption twice I think, once here and once again when it is caught on line 57?

Reply:
True, will remove one

Reply:
Fixed